voxblox_ros/src/intensity_server.cc

```diff
@@ -0,0 +1,123 @@
+#include "voxblox_ros/intensity_server.h"
+
+namespace voxblox {
+
+IntensityServer::IntensityServer(const ros::NodeHandle& nh,
+                                 const ros::NodeHandle& nh_private)
+    : TsdfServer(nh, nh_private),
+      focal_length_px_(391.5f),
+      subsample_factor_(12) {
+  cache_mesh_ = true;
+
+  intensity_layer_.reset(
+      new Layer<IntensityVoxel>(tsdf_map_->getTsdfLayer().voxel_size(),
+                                tsdf_map_->getTsdfLayer().voxels_per_side()));
+  intensity_integrator_.reset(new IntensityIntegrator(tsdf_map_->getTsdfLayer(),
+                                                      intensity_layer_.get()));
+
+  // Get ROS params:
+  nh_private_.param("intensity_focal_length", focal_length_px_,
+                    focal_length_px_);
+  nh_private_.param("subsample_factor", subsample_factor_, subsample_factor_);
+
+  float intensity_min_value = 10.0;
+  float intensity_max_value = 40.0;
+  nh_private_.param("intensity_min_value", intensity_min_value,
+                    intensity_min_value);
+  nh_private_.param("intensity_max_value", intensity_max_value,
+                    intensity_max_value);
+
+  FloatingPoint intensity_max_distance =
+      intensity_integrator_->getMaxDistance();
+  nh_private_.param("intensity_max_distance", intensity_max_distance,
+                    intensity_max_distance);
+  intensity_integrator_->setMaxDistance(intensity_max_distance);
+
+  // Publishers for output.
+  intensity_pointcloud_pub_ =
+      nh_private_.advertise<pcl::PointCloud<pcl::PointXYZI> >(
+          "intensity_pointcloud", 1, true);
+  intensity_mesh_pub_ =
+      nh_private_.advertise<voxblox_msgs::Mesh>("intensity_mesh", 1, true);
+
+  color_map_.reset(new IronbowColorMap());
+  color_map_->setMinValue(intensity_min_value);
+  color_map_->setMaxValue(intensity_max_value);
+
+  // Set up subscriber.
+  intensity_image_sub_ = nh_private_.subscribe(
+      "intensity_image", 1, &IntensityServer::intensityImageCallback, this);
+}
+
+void IntensityServer::updateMesh() {
+  TsdfServer::updateMesh();
+
+  // Now recolor the mesh...
+  timing::Timer publish_mesh_timer("intensity_mesh/publish");
+  recolorVoxbloxMeshMsgByIntensity(*intensity_layer_, color_map_, &mesh_msg_);
```

Comment:
Oh I see, so you just overwrite the cached mesh of the TsdfServer. I guess that's probably more efficient than repainting and then reconverting the underlying voxblox mesh type.